examples/depthEstimation-single-image/sketch.js

```diff
@@ -0,0 +1,137 @@
+        imgWidth = img.width;
+        imgHeight = img.height;
+        resizeCanvas(imgWidth * 2, imgHeight); // Resize canvas now
+
+        // Wait for the model to be ready
+        status = "Waiting for model to load...";
+        console.log(status);
+        await depthEstimator.ready; // Wait for the promise returned by the constructor
+        console.log("Model loaded successfully!");
+
+        // Now both image and model are ready, start estimation
+        estimateDepth();
+
+    } catch (error) {
+        console.error("Error loading image:", error);
+        status = "Error loading image. Check console.";
+    }
+}
+
+// Helper function to load image with a Promise
+function loadImagePromise(path) {
+    return new Promise((resolve, reject) => {
+        loadImage(path, img => {
+            if (img) {
+                resolve(img);
+            } else {
+                reject(`Failed to load image at path: ${path}`);
+            }
+        }, err => {
+            reject(`Error loading image: ${err}`);
+        });
+    });
+}
+
+
+// Function to start the depth estimation
+async function estimateDepth() {
+    if (!img || !depthEstimator || !depthEstimator.model) { // Add check for depthEstimator.model
+        console.error("Estimation called before model or image was ready.");
+        status = "Error: Model or image not ready.";
+        return;
+    }
+    status = "Estimating depth...";
+    console.log(status);
+    // Estimate depth from the loaded image
+    // Use await here as estimate is async
+    depthResult = await depthEstimator.estimate(img);
+    gotResults(depthResult); // Call gotResults directly after await
+}
+
+
+function draw() {
+    background(0); // Clear the background
+
+    // Draw the original image on the left half of the canvas
+    if (img) {
+        image(img, 0, 0, imgWidth, imgHeight);
+    } else {
+        // Show status if image hasn't loaded yet
+        fill(255);
+        noStroke();
+        textAlign(CENTER, CENTER);
+        textSize(16);
+        text(status, imgWidth / 2, imgHeight / 2);
+        textAlign(LEFT, BASELINE); // Reset text alignment
+    }
+
+    // Check if depth estimation results are available
+    if (depthResult) {
+        // Draw the colormapped depth visualization on the right half of the canvas
+        image(depthResult.visualizationImage, imgWidth, 0, imgWidth, imgHeight);
+    } else {
+        // If no results yet, display the status message on the right half
+        fill(255);
+        noStroke();
+        textAlign(CENTER, CENTER);
+        textSize(16);
+        text(status, imgWidth + imgWidth / 2, imgHeight / 2);
+        textAlign(LEFT, BASELINE); // Reset text alignment
+    }
+}
+
+// Callback function that receives the depth estimation results
+function gotResults(result) {
+    // Store the result in the global variable (already done by await in estimateDepth)
+    // depthResult = result;
```

Comment:
With `await`, it might be clearer to forgo the "callback" function entirely, and do it all in the caller?